client/src/components/Loading.tsx

```diff
@@ -1,10 +1,23 @@
+import { Suspense } from "react";
+import { Await, useLoaderData } from "react-router-dom";
 import { Box, CircularProgress } from "@chakra-ui/react";
+import { Login } from "./pages";
 
 const Loading = () => {
+  const data = useLoaderData() as { response: null };
+
   return (
-    <Box bg="#156087" display="flex" minH="calc(100vh)">
-      <CircularProgress isIndeterminate color="white" m="auto" />
-    </Box>
+    <Suspense
+      fallback={
+        <Box bg="#156087" display="flex" minH="100vh">
+          <CircularProgress isIndeterminate color="orange" m="auto" />
+        </Box>
+      }
+    >
+      <Await resolve={data.response} errorElement={<Login />}>
```

Comment:
Maybe instead of the error element being `<Login />`, we could redirect them to the `/login` route using the `<Navigate />` component .

https://reactrouter.com/en/main/components/navigate